src/units.cpp

```diff
@@ -370,12 +370,12 @@ void Units::unitAttributes(size_t index, std::string &reference, std::string &pr
     reference = u.mReference;
     prefix = u.mPrefix;
     if (!u.mExponent.empty()) {
-        exponent = convertToDouble(u.mExponent);
+        convertToDouble(u.mExponent, exponent);
     } else {
```

Comment:
If the conversion to double fails, what do you expect the exponent to be set to? In one line it's defaulting to 1.0, in the other it's failing silently and giving an unknown value to the returned exponent ... I think it might be better to test the return bool here too, or is that caught somewhere else ...?

Reply:
Good question! It's not my code in the first place, so I merely updated it, that's all. I mean, the very original code was `exponent = std::stod(u.mExponent);`, so I am assuming that at this stage we know that `u.mExponent` is valid...?

Ok, I am going to double check this...

Reply:
Ok, that method is used in a lot of places. So, if the conversion fails, we now default to `1.0` (see commit 529b2ab5). Not sure if that's the best way to go about this though. @hsorby?

Reply:
In line with the earlier design changes, I don't think it should be defaulting to anything!  I do think that the `false` return should be caught and passed back to the calling function for error handling though ... ?

Reply:
I would probably agree with that, but that method is used so many times that I would rather leave it to the person who wrote that code in the first place. Also, this goes beyond the purpose of this PR.